frontends/web/src/routes/device/components/skipfortesting.tsx

```diff
@@ -16,32 +16,45 @@
 
 import React, { useState } from 'react';
 import { testRegister } from '../../../api/account';
+import { getTesting } from '../../../api/backend';
 import { Button } from '../../../components/forms';
 import { PasswordSingleInput } from '../../../components/password';
+import { Dialog, DialogButtons } from '../../../components/dialog/dialog';
+import { useLoad } from '../../../hooks/api';
+import { debug } from '../../../utils/env';
 
 export const SkipForTesting = () => {
+  const [dialog, setDialog] = useState(false);
+  const show = useLoad(debug ? getTesting : () => Promise.resolve(false));
   const [testPIN, setTestPIN] = useState('');
   const registerTestingDevice = async (e: React.SyntheticEvent) => {
     e.preventDefault();
     await testRegister(testPIN);
+    setDialog(false);
   };
-  const handleFormChange = (value: string) => {
-    setTestPIN(value);
-  };
-
 
+  if (!show) {
+    return null;
+  }
+  const title = 'Unlock software kestore';
   return (
-    <form onSubmit={registerTestingDevice} style={{ flexGrow: 0, maxWidth: 400, width: '100%', alignSelf: 'center' }}>
-      <PasswordSingleInput
-        type="password"
-        autoFocus
-        label="Test Password"
-        onValidPassword={handleFormChange}
-        value={testPIN} />
-      <Button type="submit" secondary>
-        Skip for Testing
-      </Button>
-    </form>
+    <>
+      <button onClick={() => setDialog(true)}>{title}</button>
+      <Dialog open={dialog} title={title} onClose={() => setDialog(false)}>
+        <form onSubmit={registerTestingDevice}>
+          <PasswordSingleInput
+            type="password"
+            autoFocus
```

Comment:
@thisconnect when i open the dialog the input is not auto-focused. any idea why?

Reply:
lol, the checkbox has focus, this is probably because we have some old weird logic in the dialog that tries to focus the first input element. This was made when we always had focus issues in preact. In react I noticed almost no more focus issues. It's probably time to refactor dialog.tsx. I guess we can just remove the old logic, here: https://github.com/digitalbitbox/bitbox-wallet-app/blob/2c667fd98b8f1261c1d7767e983fb7258e8f11aa/frontends/web/src/components/dialog/dialog.tsx#L83-L101